fedimint-client/src/client/builder.rs

```diff
@@ -393,42 +395,65 @@ impl ClientBuilder {
         }
 
         let stopped = self.stopped;
-        self.build(pre_root_secret, config, api_secret, stopped)
-            .await
+        self.build(
+            pre_root_secret,
+            config,
+            api_secret,
+            stopped,
+            preview_prefetch_api_announcements,
+        )
+        .await
     }
 
-    pub async fn preview(mut self, invite_code: &InviteCode) -> anyhow::Result<ClientPreview> {
+    pub async fn preview(self, invite_code: &InviteCode) -> anyhow::Result<ClientPreview> {
         let (config, api) = self
             .connector
             .download_from_invite_code(invite_code, self.iroh_enable_dht, self.iroh_enable_next)
             .await?;
 
-        if let Some(guardian_pub_keys) = config.global.broadcast_public_keys.clone() {
-            // Fetching api announcements using invite urls before joining, then write them
-            // to database This ensures the client can communicated with the
-            // Federation even if all the peers moved.
-            refresh_api_announcement_sync(&api, self.db_no_decoders(), &guardian_pub_keys).await?;
-        }
-
-        self.reuse_connector = Some(api.connector().clone());
+        let prefetch_api_announcements =
+            config
+                .global
+                .broadcast_public_keys
+                .clone()
+                .map(|guardian_pub_keys| {
+                    JitTry::new_try({
+                        let db = self.db_no_decoders().clone();
+                        let api = api.clone();
+                        || async move {
+                            // Fetching api announcements using invite urls before joining, then
+                            // write them to database This ensures the
+                            // client can communicated with the
+                            // Federation even if all the peers moved.
+                            refresh_api_announcement_sync(&api, &db, &guardian_pub_keys).await
```

Comment:
can we do this in two parts: fetch annoucement in jit and store it in db during joining. this would remove db required by ClientPreview.

Reply:
no require db in client preview, so fedi can use it - we don't have a db allocated during preview phase.

Reply:
@maan2003 https://github.com/fedimint/fedimint/pull/7830